Failing test: Security Solution Cypress.x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/alerts_charts·cy·ts - KPI visualizations in Alerts Page Histogram legend hover actions should add To Timeline should add To Timeline

A test failed on a tracked branch

```
AssertionError: Timed out retrying after 150000ms: Expected to find element: `[data-test-subj="alerts-histogram-panel"] .echLegendItem__action`, but never found it.
    at clickAlertsHistogramLegend (webpack:///./tasks/alerts.ts:401:5)
    at Context.eval (webpack:///./e2e/investigations/alerts/alerts_charts.cy.ts:135:33)
```

First failure: [kibana-on-merge - main](https://buildkite.com/elastic/kibana-on-merge/builds/78721#0199bbe3-8fa1-4d79-8216-b8a416c3bc2a)

<!-- kibanaCiData = {"failed-test":{"test.class":"Security Solution Cypress.x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/alerts_charts·cy·ts","test.name":"KPI visualizations in Alerts Page Histogram legend hover actions should add To Timeline should add To Timeline","test.failCount":1}} -->

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/alerts_charts.cy.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/alerts_charts.cy.ts
@@ -109,7 +109,7 @@ describe('KPI visualizations in Alerts Page', { tags: ['@ess', '@serverless'] },
     it('should should add a filter in to KQL bar', () => {
       selectAlertsHistogram();
       const expectedNumberOfAlerts = 1;
-      clickAlertsHistogramLegend();
+      clickAlertsHistogramLegend(ruleConfigs.name);
       clickAlertsHistogramLegendFilterFor(ruleConfigs.name);
       cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM).should(
         'have.text',
@@ -120,7 +120,7 @@ describe('KPI visualizations in Alerts Page', { tags: ['@ess', '@serverless'] },
 
     it('should add a filter out to KQL bar', () => {
       selectAlertsHistogram();
-      clickAlertsHistogramLegend();
+      clickAlertsHistogramLegend(ruleConfigs.name);
       clickAlertsHistogramLegendFilterOut(ruleConfigs.name);
       cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM).should(
         'have.text',
@@ -134,7 +134,7 @@ describe('KPI visualizations in Alerts Page', { tags: ['@ess', '@serverless'] },
 
     it('should add To Timeline', () => {
       selectAlertsHistogram();
-      clickAlertsHistogramLegend();
+      clickAlertsHistogramLegend(ruleConfigs.name);
       clickAlertsHistogramLegendAddToTimeline(ruleConfigs.name);
 
       cy.get(TOASTER).should('have.text', `Added ${ruleConfigs.name} to Timeline`);

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/screens/alerts.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/screens/alerts.ts
@@ -149,10 +149,10 @@ export const ALERT_SUMMARY_CHARTS_COLLAPSED = getDataTestSubjectSelector('chart-
 
 export const ALERTS_HISTOGRAM = getDataTestSubjectSelector('alerts-histogram-panel');
 
-export const ALERTS_HISTOGRAM_LEGEND =
-  '[data-test-subj="alerts-histogram-panel"] .echLegendItem__action';
-
 export const ALERTS_HISTOGRAM_SERIES = '[data-ech-series-name]';
+
+export const ALERTS_HISTOGRAM_LEGEND_BUTTON = (ruleName: string) =>
+  `[data-test-subj="alerts-histogram-panel"] [data-test-subj="legend-${ruleName}"]`;
 
 export const SELECT_HISTOGRAM = '[data-test-subj="chart-select-trend"]';
 

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/alerts.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/alerts.ts
@@ -25,7 +25,9 @@ import {
   ALERT_TAGGING_CONTEXT_MENU,
   ALERT_TAGGING_CONTEXT_MENU_ITEM,
   ALERT_TAGGING_UPDATE_BUTTON,
-  ALERTS_HISTOGRAM_LEGEND,
+  ALERTS_HISTOGRAM,
+  ALERTS_HISTOGRAM_LEGEND_BUTTON,
+  ALERTS_HISTOGRAM_SERIES,
   ALERTS_TABLE_ROW_LOADER,
   CELL_ADD_TO_TIMELINE_BUTTON,
   CELL_FILTER_IN_BUTTON,
@@ -394,8 +396,19 @@ export const openAnalyzerForFirstAlertInTimeline = () => {
   cy.get(OPEN_ANALYZER_BTN).first().click({ force: true });
 };
 
-export const clickAlertsHistogramLegend = () => {
-  cy.get(ALERTS_HISTOGRAM_LEGEND).click();
+export const clickAlertsHistogramLegend = (ruleName: string) => {
+  cy.get(ALERTS_HISTOGRAM)
+    .find(ALERTS_HISTOGRAM_SERIES)
+    .should('contain.text', ruleName);
+
+  recurse(
+    () => {
+      cy.get('body').type('{esc}');
+      cy.get(ALERTS_HISTOGRAM).contains(ruleName).realHover();
+      return cy.get(ALERTS_HISTOGRAM_LEGEND_BUTTON(ruleName));
+    },
+    ($el) => $el.is(':visible')
+  ).click();
 };
 
 export const clickAlertsHistogramLegendAddToTimeline = (ruleName: string) => {

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/alerts.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/alerts.ts
@@ -397,9 +397,7 @@ export const openAnalyzerForFirstAlertInTimeline = () => {
 };
 
 export const clickAlertsHistogramLegend = (ruleName: string) => {
-  cy.get(ALERTS_HISTOGRAM)
-    .find(ALERTS_HISTOGRAM_SERIES)
-    .should('contain.text', ruleName);
+  cy.get(ALERTS_HISTOGRAM).find(ALERTS_HISTOGRAM_SERIES).should('contain.text', ruleName);
 
   recurse(
     () => {

